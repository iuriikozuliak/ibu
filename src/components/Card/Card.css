.Card {
  composes: container from 'theme/layout.css';
  composes: shadow from 'theme/shadow.css';
}
