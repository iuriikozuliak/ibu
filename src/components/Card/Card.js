import React     from 'react';
import PropTypes from 'prop-types';

import styles from './Card.css';

const Card = ({ children }) => (
  <div className = { styles.Card }>
    { children }
  </div>
);

Card.propTypes = {
  children: PropTypes.element.isRequired
};

export default Card;
