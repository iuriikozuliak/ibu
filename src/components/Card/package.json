{
  "name": "Card",
  "private": true,
  "main": "./Card.js"
}
