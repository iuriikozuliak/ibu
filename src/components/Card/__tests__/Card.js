import React    from 'react';
import renderer from 'react-test-renderer';
import Card     from '../Card';

test('Renders Card component', () => {
  const component = renderer.create(
    <Card />
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('Renders Card component with passed children', () => {
  const component = renderer.create(
    <Card>  
      <h2>Card Title</h2>
    </Card>
  ).toJSON();
  expect(component).toMatchSnapshot();
});
