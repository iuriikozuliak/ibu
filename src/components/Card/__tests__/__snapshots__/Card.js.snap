// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Card component 1`] = `
<div
  className="Card"
/>
`;

exports[`Renders Card component with passed children 1`] = `
<div
  className="Card"
>
    
  <h2>
    Card Title
  </h2>
</div>
`;
